glossary.rst

```diff
@@ -5,18 +5,30 @@ Glossary
 
 .. glossary::
 
-    service
-        A service is in the context of MICO.
+    MICO Application
+        An application is a set of services.
+    
+    MICO Service
+        MICO Service is the resource that is managed by the MICO system, is part of the MICO domain model and relates to the term "micro-service". Such a MICO Service originates either from a GitHub repository or from DockerHub.
 
-    external service
-        An external service is a :term:`service` not managed by the mico system but by a third party. Such a service originates either from a GitHub repository or from DockerHub.
+    Kubernetes Service
+        Kubernetes Service "is an abstraction which defines a logical set of Pods and a policy by which to access them - sometimes called a micro-service". Pods themselves "serve as unit of deployment, horizontal scaling, and replication". (quoted by Kubernetes documentation)
+        The relationship between a MICO Service and a Kubernetes Service is quite confusing because of following association:
+            - MICO Service -> Kubernetes Deployment
+            - MICO Service Interface → Kubernetes Service (MICO Service Interface is a different resource within the mico system, but is part of a MICO Service, actually it is a 1:n releationship)
+        
+    Service Layer
+        The Service Layer is the business logic.
+        
+    Broker
+        Broker is part of the domain layer that extends the domain model with more business logic. A Broker adds operations to the model in ubiquitous language (not CRUD, that would be a Repository), but is completely stateless.
+        
+    Service
+        An application service retrieve the information required by an entity, effectively setting up the execution environment, and provide it to the entity. For example for the use-case of importing MicoServices based on GitHub repositories the Application Service GitHubCrawler exists.
 
-    service interface
+    External Service
+        An external service is a service not managed by the mico system but by a third party, e.g. a weather api from weatherbit which could be used by a MICO Service. External services are currently not supported.
+        
+    Service Interface
         A service interface is an API (REST over http, gRPC, etc.) the service provides for other services to use over the network.
-        One service can have multiple service interfaces.
-
-    application
-        An application is a set of services.
-
-    admin user
-        At the moment there is no difference between a user and an admin user.
+        One MICO service can have multiple service interfaces, but at least one.
```

Comment:
Add the information, that a MICO service interface corresponds to a Kubernetes Service.